ui/packages/shared/profile/src/ProfileMetricsGraph/index.tsx

```diff
@@ -89,13 +89,15 @@ const ProfileMetricsGraph = ({
 }: ProfileMetricsGraphProps): JSX.Element => {
   const {isLoading, response, error} = useQueryRange(queryClient, queryExpression, from, to);
   const isLoaderVisible = useDelayedLoader(isLoading);
-  const {loader} = useParcaContext();
+  const {loader, onError} = useParcaContext();
 
   if (isLoaderVisible) {
     return <>{loader}</>;
   }
 
   if (error !== null) {
+    onError?.metricsGraphError(error);
```

Comment:
@yomete Maybe we should move this call into an effect on `error` variable, as this will invoke the callback for every render of the component.